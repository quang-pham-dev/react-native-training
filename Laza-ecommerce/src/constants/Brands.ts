export const BRANDS_EMPTY_RESULT = 'No Brand found';
