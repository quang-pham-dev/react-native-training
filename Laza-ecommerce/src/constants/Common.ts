//Change to http://10.0.2.2 for android and http://localhost for iOS
export const APP_BASE_URL = 'http://localhost:3000';

// Endpoints
export const ENDPOINTS = {
  AUTH: {
    LOGIN: '/auth/login',
    LOGOUT: '/auth/logout'
  },

  PRODUCT: {
    GET_PRODUCTS: '/products',
    GET_PRODUCT: '/products/:id'
  },
  BRAND: {
    GET_BRANDS: '/brands',
    GET_BRAND: '/brands/:id'
  }
};

// Auth Data key for local storage
export const AUTH_DATA = '@AuthData';

export const BOUNDARY_ERROR_MESSAGE = 'Something went wrong!';

export const SENTRY_DSN =
  'https://30f74c15a1c54630a55c03e195d86cab@o1296237.ingest.sentry.io/6522749';
