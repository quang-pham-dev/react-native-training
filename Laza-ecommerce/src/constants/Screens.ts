export const SCREENS_ROUTES = {
  STACK: {
    APP: { name: 'AppStack' },
    AUTH: { name: 'AuthStack' },
    HOME: { name: 'HomeStack' },
    BAGS: { name: 'BagsStack' },
    WALLET: { name: 'WalletStack' },
    WISHLIST: { name: 'WishListStack' },
  },

  // routes for the auth stack
  AUTH_STACK: {
    SIGN_IN_SCREEN: { name: 'SignIn', label: 'SIGN IN' },
    SIGN_UP_SCREEN: { name: 'SignUp', label: 'SIGN UP' },
    GET_STARTED_SCREEN: { name: 'GetStarted', label: 'GET STARTED' },
  },

  // routes for the home stack
  HOME_STACK: {
    HOME_SCREEN: { name: 'Home', label: 'HOME' },
    DRAWER_MENU: { name: 'DrawerMenu', label: 'MENU' },
    PRODUCT_DETAIL_SCREEN: { name: 'ProductDetail', label: 'PRODUCT DETAIL' },
    BRAND_DETAI_SCREENL: { name: 'BrandDetail', label: 'BRAND DETAIL' },
  },

  // routes for the bags stack
  WISHLIST_STACK: {
    WISHLIST_SCREEN: { name: 'WishList', label: '   WISHLIST' },
  },
  // routes for the bags stack
  BAGS_STACK: {
    BAGS_SCREEN: { name: 'Bags', label: 'BAGS' },
  },

  // routes for the wallet stack
  WALLET_STACK: {
    WALLET_SCREEN: { name: 'Wallet', label: 'WALLET' },
  },

  APP_NAVIGATOR: { name: 'AppNavigator' },
  AUTH_NAVIGATOR: { name: 'AuthNavigator' },
};
