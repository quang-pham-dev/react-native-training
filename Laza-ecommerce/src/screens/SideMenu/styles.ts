import { StyleSheet } from 'react-native';
// Themes
import { Metrics, Fonts, Colors } from 'styles/themes';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: Metrics.padding.lg,
  },

  header: {
    flex: 1,
  },

  iconMenuWrapper: {
    flexDirection: 'row',
    marginTop: Metrics.margin.xxxl,
  },

  iconMenu: {
    width: 45,
    height: 45,
  },

  infoWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: Metrics.margin.xxl,
  },

  profileInfoWrapper: {
    justifyContent: 'space-between',
    marginLeft: Metrics.margin.xm,
  },

  profile: {
    flexDirection: 'row',
  },

  verifiedWrapper: {
    flexDirection: 'row',
  },

  verifiedText: {
    fontSize: Fonts.size.small,
    fontFamily: Fonts.fontFamily.Inter_400Regular,
    color: Colors.textGray,
  },
  accountName: {
    fontSize: Fonts.size.default,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    color: Colors.textBlack,
  },

  iconBadge: {
    width: 15,
    height: 15,
    resizeMode: 'contain',
  },

  orderInfo: {
    padding: Metrics.padding.sm,
    backgroundColor: Colors.secondaryBackground,
    borderRadius: Metrics.borderRadius.roundedSmall,
  },
  textOrder: {
    fontSize: Fonts.size.min,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    color: Colors.textGray,
  },

  avatar: {
    width: 45,
    height: 45,
    borderRadius: 45,
    resizeMode: 'cover',
  },

  icons: {
    width: 19,
    height: 19,
    resizeMode: 'contain',
    marginRight: Metrics.margin.md,
  },

  text: {
    fontSize: Fonts.size.normal,
    fontFamily: Fonts.fontFamily.Inter_400Regular,
    lineHeight: Fonts.lineHeight.base,
  },

  main: {
    flex: 2,
  },
  darkModeWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: Metrics.margin.xl,
  },

  darkMode: {
    flexDirection: 'row',
    alignItems: 'center',
  },

  switchWrapper: {
    alignItems: 'center',
  },

  darkModeSwitch: {
    transform: [{ scaleX: 0.6 }, { scaleY: 0.6 }],
  },

  boxWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: Metrics.margin.xl,
  },

  footer: {
    flex: 1,
    justifyContent: 'center',
  },

  logoutText: {
    fontSize: Fonts.size.normal,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    color: Colors.textLogout,
  },
});

export default styles;
