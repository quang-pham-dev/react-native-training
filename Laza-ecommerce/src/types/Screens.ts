import { UserSignIn } from 'types/User';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { AuthStackParams, HomeStackParams } from 'types/Navigation';
import { ProductProps } from './Products';

export type AuthNavigationProp = NativeStackNavigationProp<AuthStackParams>;

export interface GetStartedScreenProps {
  navigation: AuthNavigationProp;
}

export interface SignInScreenProps {
  navigation: AuthNavigationProp;
  onSubmit?: (data: UserSignIn) => void;
}

export type AppNavigationProp = NativeStackNavigationProp<HomeStackParams>;

export interface HomeScreenProps {
  navigation: {
    navigate: (value: string, params: string) => void;
    toggleDrawer: () => void;
  };
}
export interface HeaderProps {
  navigation: {
    toggleDrawer: () => void;
  };
}
export interface ProductDetailProps {
  navigation: {
    navigate: (value: string, params: string) => void;
    goBack: () => void;
  };
  route: {
    params: string;
  };
  productData: ProductProps;
}

export interface BrandDetailProps {
  navigation: {
    navigate: (value: string, params: string) => void;
    goBack: () => void;
  };
  route: {
    params: string;
  };
  productData: ProductProps;
}
