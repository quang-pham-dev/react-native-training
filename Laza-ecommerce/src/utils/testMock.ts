export const navigationMock: any = {
  addListener: jest.fn(),
  dangerouslyGetParent: jest.fn(),
  dismiss: jest.fn(),
  dispatch: jest.fn(),
  emit: jest.fn(),
  getParam: jest.fn(),
  goBack: jest.fn(),
  isFirstRouteInParent: jest.fn(),
  isFocused: jest.fn(),
  navigate: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  push: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  router: undefined,
  setParams: jest.fn(),
  route: undefined,
  state: {
    key: '',
    routeName: '',
    params: {},
    path: '',
    index: 1,
    isTransitioning: false,
    routes: []
  }
};
