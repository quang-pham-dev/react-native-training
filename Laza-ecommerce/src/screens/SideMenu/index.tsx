import React, { memo, useCallback, useContext, useState } from 'react';
import { Alert, Image, Switch, TouchableOpacity, View } from 'react-native';
import { DrawerActions } from '@react-navigation/native';

// Screens
import { SCREENS_ROUTES } from 'constants/Screens';

// Components
import Button from 'components/Button';
import Title from 'components/Title';

// Context
import { AppContext } from 'context/AppContext';
import { SIGN_OUT, SIGN_OUT_FAILED, SIGN_OUT_SUCCESS } from 'context/actions/auth.actions';

// API
import { authService } from 'api';

// Constants
import { AUTH_DATA } from 'constants/Common';

// Types
import { ISideMenuPros } from 'types/screens/Layout';

// Themes
import IMAGES from 'themes/Images';

// Styles
import styles from './styles';

// Utils
import { remove } from 'utils/localStorage';

const SideMenu = ({ navigation }: ISideMenuPros) => {
  const [isEnabled, setIsEnabled] = useState(false);
  const toggleSwitch = () => setIsEnabled(previousState => !previousState);
  const { authState, authDispatch } = useContext(AppContext);

  const orderCount = 3 | 0;

  // handel action sign out
  const handlePressLogout = useCallback(() => {
    Alert.alert('Logout!', 'Are you sure you want to logout?', [
      {
        text: 'Cancel',
        onPress: () => {},
      },
      {
        text: 'OK',
        onPress: async () => {
          navigation.toggleDrawer();

          authDispatch({ type: SIGN_OUT });
          try {
            await authService.signOut();
            await remove(AUTH_DATA);
            authDispatch({
              type: SIGN_OUT_SUCCESS,
            });
          } catch (error) {
            authDispatch({ type: SIGN_OUT_FAILED, payload: error });
            Alert.alert('Error', error.message);
          }
        },
      },
    ]);
  }, []);

  // handle Close menu
  const handleCloseMenu = useCallback(() => {
    navigation.dispatch(DrawerActions.closeDrawer());
  }, [navigation]);

  // handle action navigate to Bag screen
  const handleNavigateToBagScreen = useCallback(() => {
    navigation.navigate(SCREENS_ROUTES.STACK.BAGS.name);
  }, [navigation]);

  // handle action navigate to WishList screen
  const handleNavigateToWishlistScreen = useCallback(() => {
    navigation.navigate(SCREENS_ROUTES.STACK.WISHLIST.name);
  }, [navigation]);

  // handle action navigate to Wallet screen
  const handleNavigateToWalletScreen = useCallback(() => {
    navigation.navigate(SCREENS_ROUTES.STACK.WALLET.name);
  }, [navigation]);

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <View style={styles.iconMenuWrapper}>
          <Button
            testID='Side-menu-toggle'
            icon={IMAGES.iconMenuOpen}
            iconStyles={[styles.iconMenu]}
            onPress={handleCloseMenu}
          />
        </View>
        <View style={[styles.infoWrapper]}>
          <View style={styles.profile}>
            {authState?.currentUser?.avatar ? (
              <Image style={styles.avatar} source={{ uri: authState?.currentUser?.avatar }} />
            ) : (
              <Image style={styles.avatar} source={IMAGES.iconAvatar} />
            )}
            <View style={styles.profileInfoWrapper}>
              {authState?.currentUser?.username ? (
                <Title
                  titleStyles={styles.accountName}
                  titleName={authState?.currentUser?.username}
                />
              ) : null}
              <View style={styles.verifiedWrapper}>
                <Title titleName='Verified Profile' titleStyles={styles.verifiedText} />
                <Image style={styles.iconBadge} source={IMAGES.iconBadge} />
              </View>
            </View>
          </View>
          <View style={styles.orderInfo}>
            <Title titleName={`${orderCount} Orders`} titleStyles={styles.textOrder} />
          </View>
        </View>
      </View>
      <View style={styles.main}>
        <View style={[styles.darkModeWrapper]}>
          <View style={[styles.darkMode]}>
            <Image style={styles.icons} source={IMAGES.iconSun} />
            <Title titleName='Dark Mode' titleStyles={styles.text} />
          </View>
          <View style={styles.switchWrapper}>
            <Switch
              style={styles.darkModeSwitch}
              trackColor={{ false: '#767577', true: '#34C759' }}
              thumbColor={isEnabled ? 'white' : '#f4f3f4'}
              ios_backgroundColor='#3e3e3e'
              onValueChange={toggleSwitch}
              value={isEnabled}
            />
          </View>
        </View>
        <TouchableOpacity
          style={[styles.boxWrapper]}
          onPress={() => {}}
          testID='AccountInformation'>
          <Image style={styles.icons} source={IMAGES.iconInfo} />
          <Title titleName='Account Information' titleStyles={styles.text} />
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.boxWrapper]}
          onPress={handleNavigateToBagScreen}
          testID='Side-menu-Bag'>
          <Image style={styles.icons} source={IMAGES.iconBagDrawer} />
          <Title titleName='Order' titleStyles={styles.text} />
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.boxWrapper]}
          onPress={handleNavigateToWalletScreen}
          testID='Side-menu-Wallet'>
          <Image style={styles.icons} source={IMAGES.iconWalletDrawer} testID='Side-menu-Cards' />
          <Title titleName='My Cards' titleStyles={styles.text} />
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.boxWrapper]}
          onPress={handleNavigateToWishlistScreen}
          testID='Side-menu-Wishlist'>
          <Image style={styles.icons} source={IMAGES.iconHeartDrawer} />
          <Title titleName='Wishlist' titleStyles={styles.text} />
        </TouchableOpacity>
      </View>
      <View style={styles.footer}>
        <TouchableOpacity
          style={[styles.boxWrapper]}
          onPress={handlePressLogout}
          testID='Side-menu-Logout'>
          <Image style={styles.icons} source={IMAGES.iconLogout} />
          <Title titleName='Logout' titleStyles={styles.logoutText} />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default SideMenu;
