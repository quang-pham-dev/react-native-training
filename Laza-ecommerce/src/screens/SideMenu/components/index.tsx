import React from 'react';
import { View, Text, Pressable, Image } from 'react-native';

// Types
import { DrawerItemProps } from 'types/screens/Menu';

// Styles
import styles from './styles';

const DrawerItem = ({ title, source, onPress, titleStyle }: DrawerItemProps) => (
  <Pressable onPress={onPress}>
    <View style={styles.boxWrapper}>
      <Image source={source} style={styles.icons} />
      <Text style={[styles.text, titleStyle]}>{title}</Text>
    </View>
  </Pressable>
);

export default DrawerItem;
