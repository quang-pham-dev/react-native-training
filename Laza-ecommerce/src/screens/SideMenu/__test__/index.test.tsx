import React from 'react';
import { Pressable } from 'react-native';

// LIBS
import { DrawerActions } from '@react-navigation/native';
import { fireEvent, render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

// Screen
import SideMenu from 'screens/SideMenu';

// Constants
import { SCREENS_ROUTES } from 'constants/Screens';

// Utils
import { navigationMock } from 'utils/testMock';

describe('Side menu', () => {
  let tree: any;
  const closeMenuID = 'Side-menu-close';

  const props = {
    navigation: navigationMock,
  };
  tree = renderer.create(<SideMenu {...props} />);

  test('should render correctly', () => {
    const component = tree.toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should call function onCloseMenuHandler', async () => {
    const closeMenu = tree.root.findByProps({ testID: closeMenuID });
    await fireEvent.press(closeMenu);

    expect(props.navigation.dispatch).toHaveBeenCalledWith(DrawerActions.closeDrawer());
  });

  test('should navigated to handleNavigateToBagScreen', () => {
    const button = tree.root.findAllByType(Pressable)[2];
    button.props.onPress();

    expect(navigationMock.navigate).toHaveBeenCalledWith(SCREENS_ROUTES.STACK.BAGS.name);
  });

  test('should navigated to handleNavigateToWalletScreen', () => {
    const button = tree.root.findAllByType(Pressable)[3];
    button.props.onPress();

    expect(navigationMock.navigate).toHaveBeenCalledWith(SCREENS_ROUTES.STACK.WALLET.name);
  });

  test('should navigated to handleNavigateToWishlistScreen', () => {
    const button = tree.root.findAllByType(Pressable)[4];
    button.props.onPress();

    expect(navigationMock.navigate).toHaveBeenCalledWith(SCREENS_ROUTES.STACK.WISHLIST.name);
  });
});
