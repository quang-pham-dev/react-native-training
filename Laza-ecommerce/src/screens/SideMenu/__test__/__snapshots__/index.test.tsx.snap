// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side menu should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 45,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 45,
            "opacity": 1,
            "width": 45,
          }
        }
      >
        <Image
          source={1}
          style={Object {}}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 30,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "borderRadius": 45,
              "height": 45,
              "resizeMode": "cover",
              "width": 45,
            }
          }
        />
        <View
          style={
            Object {
              "justifyContent": "space-between",
              "marginLeft": 15,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Inter_500Medium",
                  "fontSize": 17,
                  "lineHeight": 16,
                  "textTransform": "none",
                },
                undefined,
                Object {
                  "textAlign": "left",
                },
              ]
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "Inter_500Medium",
                    "fontSize": 17,
                    "lineHeight": 16,
                    "textTransform": "none",
                  },
                  Object {
                    "color": "#8F959E",
                    "fontFamily": "Inter_400Regular",
                    "fontSize": 13,
                  },
                  Object {
                    "textAlign": "left",
                  },
                ]
              }
            >
              Verified Profile
            </Text>
            <Image
              source={1}
              style={
                Object {
                  "height": 15,
                  "resizeMode": "contain",
                  "width": 15,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#F5F6FA",
            "borderRadius": 5,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Inter_500Medium",
                "fontSize": 17,
                "lineHeight": 16,
                "textTransform": "none",
              },
              Object {
                "color": "#8F959E",
                "fontFamily": "Inter_500Medium",
                "fontSize": 11,
              },
              Object {
                "textAlign": "left",
              },
            ]
          }
        >
          3 Orders
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginTop": 25,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 19,
                "marginRight": 13,
                "resizeMode": "contain",
                "width": 19,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Inter_400Regular",
                  "fontSize": 15,
                  "lineHeight": 16,
                },
                undefined,
              ]
            }
          >
            Dark Mode
          </Text>
        </View>
      </View>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#34C759"
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Array [
              Object {
                "marginBottom": -20,
                "transform": Array [
                  Object {
                    "scaleX": 0.6,
                  },
                  Object {
                    "scaleY": 0.6,
                  },
                ],
              },
              Object {
                "backgroundColor": "#F2F2F2",
                "borderRadius": 16,
              },
            ],
          ]
        }
        thumbTintColor="#F2F2F2"
        tintColor="#F2F2F2"
        value={false}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 25,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 19,
              "marginRight": 13,
              "resizeMode": "contain",
              "width": 19,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Inter_400Regular",
                "fontSize": 15,
                "lineHeight": 16,
              },
              undefined,
            ]
          }
        >
          Account information
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 25,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 19,
              "marginRight": 13,
              "resizeMode": "contain",
              "width": 19,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Inter_400Regular",
                "fontSize": 15,
                "lineHeight": 16,
              },
              undefined,
            ]
          }
        >
          Order
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 25,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 19,
              "marginRight": 13,
              "resizeMode": "contain",
              "width": 19,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Inter_400Regular",
                "fontSize": 15,
                "lineHeight": 16,
              },
              undefined,
            ]
          }
        >
          My Cards
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 25,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 19,
              "marginRight": 13,
              "resizeMode": "contain",
              "width": 19,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Inter_400Regular",
                "fontSize": 15,
                "lineHeight": 16,
              },
              undefined,
            ]
          }
        >
          WishList
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 25,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 19,
              "marginRight": 13,
              "resizeMode": "contain",
              "width": 19,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Inter_400Regular",
                "fontSize": 15,
                "lineHeight": 16,
              },
              Object {
                "color": "#FF5757",
                "fontFamily": "Inter_500Medium",
                "fontSize": 15,
              },
            ]
          }
        >
          Logout
        </Text>
      </View>
    </View>
  </View>
</View>
`;
